tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -47,38 +48,42 @@ def restore(self, keys) -> dict:
         vals = self._merge_items(first_items, remaining_items)
         return vals
 
-    def _get_items(self, keys: list) -> object:
-        items = []
+    def chunk_keys(self, keys: list) -> list:
+        """Generates the cand keys list to be used to read from DynamoDB"""
+        # have a for loop here for all the key chunks we want to go over
+        cand_keys_chunks = []
         for i in range(0, len(keys), 100):
-            count = 0
-            cand_keys = keys[i : min(len(keys), i + 100)]
-            while True:
-                resp = self.client.batch_get_item(
-                    RequestItems={
-                        self.name: {
-                            "Keys": cand_keys,
-                            "ConsistentRead": True,
-                        },
-                    },
-                )
-                items.extend(resp["Responses"][self.name])
-                if resp["UnprocessedKeys"].get(self.name) and count < 10:
-                    cand_keys = resp["UnprocessedKeys"][self.name]["Keys"]
-                    count += 1
-                elif count >= 10:
-                    error = Exception(
-                        f"tron_dynamodb_restore_failure: failed to retrieve items with keys \n{cand_keys}\n from dynamodb\n{resp}"
+            # chunks of 100 keys will be in this list
+            cand_keys_chunks.append(keys[i : min(len(keys), i + 100)])
+        return cand_keys_chunks
+
+    def _get_items(self, table_keys: list) -> object:
+        items = []
+        # precompute the cand_keys and then all we gotta do is submit stuff to the thread pool using the precomputed keys
+        cand_keys_list = self.chunk_keys(table_keys)
+        count = 0
+        while count < len(cand_keys_list):
+            with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
+                responses = [
+                    executor.submit(
+                        self.client.batch_get_item, RequestItems={self.name: {"Keys": key, "ConsistentRead": True}}
                     )
-                    raise error
-                else:
-                    break
+                    for key in cand_keys_list
```

Comment:
suggestion i would renaming the variables here a bit - it took me a while to realize that `key` here is not a single key but a chunk of keys 

maybe something like:
```python
with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
    futures = [
        executor.submit(
            self.client.batch_get_item, RequestItems={self.name: {"Keys": chunked_keys, "ConsistentRead": True}}
        )
        for chunked_keys in self.chunk_keys(table_keys)
    ]
```

Reply:
ah, we can't inline the self.chunk_keys() bit 'cause we update cand_keys_list below...

Reply:
ah, actually - i think we'd have to update table_keys, but this would otherwise work the same :)